src/components/accordion-card/accordion-card.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import Button from 'components/button';
+import Icon from 'components/icon';
+import chevronIcon from 'assets/icons/chevron.svg';
+import styles from './accordion-card-styles.scss';
+
+class AccordionCard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { isOpen: false };
+  }
+
+  componentDidMount() {
+    const { isOpen } = this.props;
+    this.setState({ isOpen });
+  }
+
+  render() {
+    const { icon, title, children, theme } = this.props;
+    const { isOpen } = this.state;
+    return (
+      <div
+        className={cx(styles.cardContainer, theme.cardContainer, {
+          [styles.open]: isOpen
+        })}
+      >
+        <div className={cx(styles.headerContainer, theme.headerContainer)}>
+          <h2 className={cx(styles.groupTitle, theme.groupTitle)}>
+            {title}
+          </h2>
+          <Button
+            onClick={() => this.setState({ isOpen: !isOpen })}
+            theme={{ button: styles.toggleButton }}
+          >
+            {
+              icon &&
+                (
+                  <Icon
+                    icon={icon}
+                    theme={{ icon: isOpen ? styles.icon : styles.iconCollapse }}
+                  />
+                )
+            }
+          </Button>
+        </div>
+        {children}
+      </div>
+    );
+  }
+}
+
+AccordionCard.propTypes = {
+  /** Initial state of the accordion */
+  isOpen: PropTypes.bool,
+  /** Icon to be displayed to represent toggle */
+  icon: PropTypes.shape({}),
```

Comment:
we could export the shapes propTypes of the reusable components and import them in cases like this, what do you think?

Reply:
This icon prop is not referring to the `<Icon/>` component but to the imported, webpack generated, `svg` so the shape may change depending on the config. I've used `shape` instead of `object` to avoid linter error `Prop type `object` is forbidden (react/forbid-prop-types)`.
Or maybe I'm not getting your point